File: zkext/trunk/src/com/ff/ui/tree/TreeItem.java
package com.ff.ui.tree;

import com.ff.Component;
import com.ff.annotation.Property;

public class TreeItem extends Component{

	@Property 
	private String itemId;
	@Property	
	private String text;
	@Property
	private Boolean expanded;
	@Property
	private Boolean root;
	@Property
	private Boolean leaf;
	@Property
	private Boolean expandable;
	@Property
	private Boolean checked;
	@Property
	private String icon;
	
	public String getText() {
		return text;
	}
	public void setText(String text) {
		this.text = text;
	}
	public Boolean getExpanded() {
		return expanded;
	}
	public void setExpanded(Boolean expanded) {
		this.expanded = expanded;
	}
	public Boolean isRoot() {
		return root;
	}
	public void setRoot(Boolean root) {
		this.root = root;
	}
	public Boolean getLeaf() {
		return leaf;
	}
	public void setLeaf(Boolean leaf) {
		this.leaf = leaf;
	}
	public Boolean getExpandable() {
		return expandable;
	}
	public void setExpandable(Boolean expandable) {
		this.expandable = expandable;
	}
	public Boolean getChecked() {
		return checked;
	}
	public void setChecked(Boolean checked) {
		this.checked = checked;
	}
	public String getIcon() {
		return icon;
	}
	public void setIcon(String icon) {
		this.icon = icon;
	}
	public String getItemId() {
		return itemId;
	}
	public void setItemId(String itemId) {
		this.itemId = itemId;
	}
	
	
}

File: zkext/trunk/src/web/js/zkext/ui/tree/TreeItem.js
/* TreeItem.js*/

zkext.ui.tree.TreeItem = zk.$extends(zkext.ui.Component,{	 
	$define: {	 
		 expandable:function(val){
			 this.setProperty("expandable","setExplandable",val);
		 }, 
		 expanded:function(val){
			 this.setProperty("expanded","setExpanded",val);
		 },
		 text:function(val){
			 this.setProperty("text","setText",val);
		 },
		 icon:function(val){
			 this.setProperty("icon","setIcon",val);
		 },
		 leaf:function(val){
			 this.setProperty("leaf","setLeaf",val);
		 },
		 itemId:function(val){
			 
		 }
		 
	},
		 	
	configure_:function(){
		this.$supers('configure_');		
		var config = this.getInitialConfig();					 
	},
		 
	toObject:function(){
		return this.toObjectRecursive(this);
	},
	
	toObjectRecursive:function(comp){
		var o = new Object();
		o.expandable = comp.getExpandable();
		o.expanded = comp.getExpanded();
		o.text = comp.getText();
		o.icon = comp.getIcon();
		o.leaf = comp.getLeaf();
		o.id = comp.uuid;
		o.children = new Array();			
		for (var w = comp.firstChild;w;w=w.nextSibling) {		
			if(zkext.ui.tree.TreeItem.isInstance(w)){
				var child = w.toObject();
				o.children.push(child);
			}
		}
		return o;
	}
			
});